### DISCUSSION TITLE
How to use https with nginx?

### DISCUSSION THREAD

**Discussion by lohrumipsum** [coder/code-server#3709] [Q&A]
Hi, 
I followed the directions in the setup guide and installed code-server on a raspberry pi.  When I go to localhost:8080 on the pi, all looks good.  My end goal is to access code server via my ipad, so I set up nginx.  I followed the instructions and I *think* I have everything set up correctly.  However, when I launch code-server, I see the following line:

[2021-07-01T23:42:49.479Z] info    - Not serving HTTPS 

How do I enable https?  VS Code works pretty well in the browser, but the web previews don't work and there are a couple of other things which seem to be impacted by the lack of https.

I apologize in advance, my skill level isn't very high.

**Comment by jsjoeio** [comment]
Hey! Thanks for the details. I have not followed this setup myself, but there is a section about setting up Let's Encrypt to [serve via HTTPS](https://github.com/cdr/code-server/blob/main/docs/guide.md#lets-encrypt). Did you try that?

**Reply by lohrumipsum** [reply]
I followed the instructions to install nginx; I thought that was the gist of that section (to install Caddy or nginx).  Did I miss a separate step somewhere?

**Reply by jsjoeio** [reply]
Right, but did you also follow the Let's Encrypt section? I think that's required to serve via HTTPS and access on your iPad? (I could be wrong though) cc @code-asher

**Reply by code-asher** [reply]
Yup you'll need to follow all the steps 1-5.

**Reply by lohrumipsum** [reply]
Thanks. I started over from scratch the other day and used certbot to create a certificate. That got the https working. As a heads-up, I got a redirect error until I changed the http: in this line to an https:  
![57404D6F-690F-438D-AA07-8A06F08A049C](https://user-images.githubusercontent.com/14303860/125363826-56954500-e33f-11eb-8861-127ecdd95e12.jpeg)

**Reply by jsjoeio** [reply]
oh thanks for the heads-up! I'll submit a PR for this

**Reply by jsjoeio** [reply]
https://github.com/cdr/code-server/pull/3768
